Jenkinsfile

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2016-present Sonatype, Inc. All rights reserved.
+ * Includes the third-party code listed at http://links.sonatype.com/products/nexus/attributions.
+ * "Sonatype" is a trademark of Sonatype, Inc.
+ */
+@Library('ci-pipeline-library') _
+import com.sonatype.jenkins.pipeline.GitHub
+import com.sonatype.jenkins.pipeline.OsTools
+
+properties([
+  parameters([
+    booleanParam(defaultValue: false, description: 'Force Red Hat Certified Build for a non-master branch', name: 'force_red_hat_build'),
+    booleanParam(defaultValue: false, description: 'Skip Red Hat Certified Build', name: 'skip_red_hat_build'),
+  ])
+])
+
+node('ubuntu-zion') {
+  def commitId, commitDate, version, isMaster
+  def organization = 'sonatype',
+      gitHubRepository = 'operator-nxiq',
+      credentialsId = 'integrations-github-api',
+      archiveName = 'nxiq-operator-certified-metadata.zip'
+  GitHub gitHub
+
+  stage('Preparation') {
+    deleteDir()
+
+    def checkoutDetails = checkout scm
+
+    isMaster = checkoutDetails.GIT_BRANCH in ['origin/master', 'master']
+
+    commitId = checkoutDetails.GIT_COMMIT
+    commitDate = OsTools.runSafe(this, "git show -s --format=%cd --date=format:%Y%m%d-%H%M%S ${commitId}")
+
+    OsTools.runSafe(this, 'git config --global user.email sonatype-ci@sonatype.com')
+    OsTools.runSafe(this, 'git config --global user.name Sonatype CI')
+
+    version = readVersion()
+
+    withCredentials([[$class: 'UsernamePasswordMultiBinding', credentialsId: credentialsId,
+                      usernameVariable: 'GITHUB_API_USERNAME', passwordVariable: 'GITHUB_API_PASSWORD']]) {
+      gitHub = new GitHub(this, "${organization}/${gitHubRepository}", env.GITHUB_API_PASSWORD)
+    }
+  }
+
+  stage('Trigger Red Hat Certified Image Build') {
+    if ((! params.skip_red_hat_build) && (isMaster || params.force_red_hat_build)) {
+      withCredentials([
+          string(credentialsId: 'operator-nxiq-rh-build-project-id', variable: 'PROJECT_ID'),
+          string(credentialsId: 'rh-build-service-api-key', variable: 'API_KEY')]) {
+        runGroovy('ci/TriggerRedHatBuild.groovy', "'${version}' '${PROJECT_ID}' '${API_KEY}'")
+      }
+    }
+  }
+
+  if (currentBuild.result == 'FAILURE') {
+    return
+  }
+
+  stage('Build') {
+    gitHub.statusUpdate commitId, 'pending', 'build', 'Build is running'
+
+    OsTools.runSafe(this, 'scripts/bundle.sh')
```

Comment:
ok, I tested a bit making these scripts fail, and they bail right out and fail...no chance to to status stuff, and from reading, maybe/probably not even setting the currentBuild.result anyway.  Anyone mind if I drop out the status stuff and the currentBuild.result stuff completely. I wanted to keep it in case it does something, but I don't think it does for the bits I've kept in this build.

Reply:
the status stuff isn't even really reporting anything useful, since the Build step is the simple bundle of some files. it takes moments.

Reply:
> ok, I tested a bit making these scripts fail, and they bail right out and fail...no chance to do status stuff, and from reading, maybe/probably not even setting the currentBuild.result anyway. Anyone mind if I drop out the status stuff and the currentBuild.result stuff completely. I wanted to keep it in case it does something, but I don't think it does for the bits I've kept in this build.

So any of the scripts called this way can fail and there's no outward indication?

Reply:
> the status stuff isn't even really reporting anything useful, since the Build step is the simple bundle of some files. it takes moments, so i'd like to just drop out all the github status stuff.

Works for me, status updates don't seem particularly valuable for this build

Reply:
they have non-zero returns from `sh`, so they manifest as an exception/failure here. I added a `false` in the shell command in this run: https://jenkins.ci.sonatype.dev/job/integrations/job/cloud/job/operator-iq/job/feature-snapshots/job/INT-2547-jenkins/7/console

Reply:
I've removed github status stuff.